pkg/controller/nodetune/status.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2024 ScyllaDB
+
+package nodetune
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+func (ncdc *Controller) calculateStatus(nc *v1alpha1.NodeConfig) *v1alpha1.NodeConfigStatus {
+	status := nc.Status.DeepCopy()
+	status.ObservedGeneration = nc.Generation
```

Comment:
This updates ObservedGeneration of entire NodeConfig, while this controller is only responsible for single Node. 
This field of NodeConfigStatus should be (and is already) reconciled by nodeconfig controller.

Reply:
I see your point, but from the correctness perspective, is there anything inherently wrong about this?

If so it should also be fixed in the existing nodesetup controller.

Reply:
ObservedGeneration shall only be updated by the main loop and relates to its fields that it protects. For fields updated from different loops you either need a separate ObservedGeneration (like for conditions) or to live without the protections it provides.

Reply:
done